posthog/temporal/data_imports/pipelines/sql_database/arrow_helpers.py

```diff
@@ -92,6 +92,13 @@ def row_tuples_to_arrow(rows: Sequence[RowAny], columns: TTableSchemaColumns, tz
                 py_type = type(val)
                 break
 
+        # Handle infinity in decimal fields
+        if pa.types.is_decimal(field.type) and issubclass(py_type, decimal.Decimal):
+            values = columnar_known_types[field.name]
+            columnar_known_types[field.name] = np.array(
+                [None if isinstance(x, decimal.Decimal) and x.is_infinite() else x for x in values]
+            )
```

Comment:
We're doing a similar thing with checking for `NaN` on line 125 - worth combining these together maybe?